packages/generator/ni_measurement_plugin_sdk_generator/client/_support.py

```diff
@@ -64,14 +64,14 @@
 ]
 
 
-def get_measurement_service_stub(
+def get_measurement_service_stub_and_version(
     discovery_client: DiscoveryClient,
     channel_pool: GrpcChannelPool,
     service_class: str,
-) -> v2_measurement_service_pb2_grpc.MeasurementServiceStub:
-    """Returns the measurement service stub of the given service class."""
+) -> Tuple[v2_measurement_service_pb2_grpc.MeasurementServiceStub, str]:
+    """Returns the measurement service stub and version of the given service class."""
     try:
-        service_location = discovery_client.resolve_service(
+        resolve_service_response = discovery_client.resolve_service_with_information(
```

Comment:
> fix: set version to empty if old discovery server is used

What does this mean? Did we previously ship a DiscoveryService that supports ResolveServiceWithInformation but which returns an empty list of versions?

@MounikaBattu17 added ResolveServiceWithInformation in June: https://github.com/ni/ni-apis/pull/51

We should either document that the client generator requires InstrumentStudio Pro 2024Q4 or whatever, or we should fall back to the old ResolveService if the new one throws RpcError with code UNIMPLEMENTED. However, the latter seems like overdoing it.

Cc: @jasonmreding

Reply:
>What does this mean? Did we previously ship a DiscoveryService that supports ResolveServiceWithInformation but which returns an empty list of versions?

Previously, there is no versions in the[ service descriptor](https://github.com/ni/ni-apis/blob/78351723cfea5f751f77dfdd8ef845a7a174c12a/ni/measurementlink/discovery/v1/discovery_service.proto#L67), when Mounika added `ResolveServiceWithInformation`. The versions are added recently in https://github.com/ni/ni-apis/pull/54/files, for versioning support.

>We should either document that the client generator requires InstrumentStudio Pro 2024Q4 or whatever

We can go with the approach of throwing click error stating, the client generator requires InstrumentStudio Pro 2024 Q4.

Reply:
Updated to throw error.

Reply:
The other option would be to remember the version from `EnumerateServices` and then pass that into the client generation rather than relying on calling `ResolveServiceWithInformation`. The advantage to this approach is the `EnumerateServices` API has always existed, it just didn't always have a `Versions` field. So if you are talking to an older version of the discovery service, this will just come back as an empty array. In that case, you could just capture the version in the client as the empty string which means it will always connect to the latest version of the service.

The down side to that approach is if you were relying on the version to indicate the generated client is incompatible with the current measurement's signature, that will no longer work.